test/blackbox-tests/test-cases/ocaml-index/gh10985.t

```diff
@@ -0,0 +1,55 @@
+This test attempts to build the ocaml index while depending on a library
+installed through a lock file.
+
+We set up a library that will be installed as part of the package:
+
+  $ mkdir external_sources
+  $ cat >external_sources/dune-project <<EOF
+  > (lang dune 3.11)
+  > (package (name mypkg))
+  > EOF
+  $ cat >external_sources/dune <<EOF
+  > (library
+  >  (public_name mypkg.lib)
+  >  (name test_lib))
+  > EOF
+  $ cat >external_sources/test_lib.ml <<EOF
+  > let x = ()
+  > EOF
+
+Now we set up a lock file with this package and then attempt to use it:
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.11)
+  > EOF
+
+  $ mkdir dune.lock
+  $ cat >dune.lock/lock.dune <<EOF
+  > (lang package 0.1)
+  > EOF
+
+  $ cat >dune.lock/mypkg.pkg <<EOF
+  > (version 0.0.1)
+  > (source (copy $PWD/external_sources))
+  > (build (run dune build --release --promote-install-file=true . @install))
+  > EOF
+
+  $ cat >dune <<EOF
+  > (dirs (:standard \ external_sources))
```

Comment:
You can probably avoid this by putting `external_sources` in one folder, the repo that uses it in another and that refer to it via `(copy $PWD/../external_sources)`.

Reply:
Right now, this file is the same as `test/blackbox-tests/test-cases/pkg/libraries.t`, just with a slightly different comment and building the index instead of `foo.cma`.  I literally started by copying over that file.  I can definitely still make this change, it just seems weird to change this from the directly analogous working test.

(If I don't change it, maybe it's worth noting this in the file, instead of just in the commit description?)

Reply:
I just made this change.  If this really matters, I can also send a separate pull request to update `pkg/libraries.t`.